pkg/migration/migration.go

```diff
@@ -111,6 +112,15 @@ func (h *Handler) Start() {
 		if err := h.processMigration(cluster, pxDaemonSet); err != nil {
 			k8sutil.WarningEvent(h.ctrl.GetEventRecorder(), cluster, util.MigrationFailedReason,
 				fmt.Sprintf("Migration failed, will retry in %v, %v", migrationRetryIntervalFunc(), err))
+
+			updatedCluster := &corev1.StorageCluster{}
+			if err := h.client.Get(context.TODO(), types.NamespacedName{Name: cluster.Name, Namespace: cluster.Namespace}, updatedCluster); err != nil {
+				logrus.Errorf("Failed to get StorageCluster. %v", err)
+			}
+			updatedCluster.Status.Phase = string(corev1.ClusterStateDegraded)
```

Comment:
Don't we need update the status of the migration condition to Failed with the error message?